Remove hardcoded reference to lm2 catalog in IT

With the addition of a new catalog `lm3`, a hardcoded reference to the `lm2` catalog in IT is causing a failure.

https://gitlab.azul.data.humancellatlas.org/ucsc/azul/-/jobs/33903
```
======================================================================
FAIL: test_indexing (integration_test.IndexingIntegrationTest) [repository_files] (catalog='lm3-it')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 378, in subTest
    yield
  File "/builds/ucsc/azul/test/integration_test.py", line 801, in _test_repository_files
    self.assertEqual(config.it_catalog_for('lm2'), catalog,
AssertionError: 'lm2-it' != 'lm3-it'
- lm2-it
?   ^
+ lm3-it
?   ^
 : Phantom files detected in catalog 'lm3-it' where none were expected
```

---

- [ ] Security design review completed; the Resolution of this issue does **not** …
  - [ ] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with 
  - [ ] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [ ] … affect the permissions of external users like access to
    - TDR snapshots
  - [ ] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [ ] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [ ] … affect monitoring of the system
  - [ ] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [ ] … add an interface that exposes sensitive or confidential data at the security boundary
  - [ ] … affect the encryption of data at rest
  - [ ] … require persistence of sensitive or confidential data that might require encryption at rest
  - [ ] … require unencrypted transmission of data within the security boundary
  - [ ] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [ ] Documentation on any unchecked boxes is provided in comments below

--- a/test/integration_test.py
+++ b/test/integration_test.py
@@ -798,8 +798,6 @@ class IndexingIntegrationTest(IntegrationTestCase, AlwaysTearDownTestCase):
                 self.assertEqual(response['Message'],
                                  f'File {file_uuid!r} with version {file_version!r} '
                                  f'was found in catalog {catalog!r}, however no download is currently available')
-                self.assertEqual(config.it_catalog_for('lm2'), catalog,
-                                 f'Phantom files detected in catalog {catalog!r} where none were expected')
             else:
                 while response['Status'] != 302:
                     self.assertEqual(301, response['Status'])

